### PR TITLE
Improve docs and add more information related with blocks

### PR DESCRIPTION
**Pull request by mrtuvn**
## Purpose of this pull request

This pull request (PR) update information related with block template, correct some minor incorrect line in doc

## Affected DevDocs pages

https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/layouts/xml-instructions.html
https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/layouts/xml-manage.html

## Links to Magento source code

-  ...

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: src/guides/v2.3/frontend-dev-guide/layouts/xml-instructions.md [hunk 1]
```diff
@@ -78,6 +78,9 @@ Sample of usage in the product listing page layout:
 <block class="Magento\Catalog\Block\Product\ListProduct" name="category.products.list" as="product_list" template="Magento_Catalog::product/list.phtml"/>
 ```
 
+{:.bs-callout-warning}
+When define block if `template` attribute not available in layout, so that block will not render by default
```
Comment (rogyar) [inline src/guides/v2.3/frontend-dev-guide/layouts/xml-instructions.md:82]:
```suggestion
When define block if `template` attribute not available in layout (or via `setTemplate()` method in the block class), that block will not render by default
```

File: src/guides/v2.3/frontend-dev-guide/layouts/xml-manage.md [hunk 2]
```diff
@@ -239,6 +239,9 @@ Example: add a block with a product [SKU](https://glossary.magento.com/sku) info
 </block>
 ```
 
+{:.bs-callout-info}
+The `template` attribute should declare with specific module: Example template="<VendorName>_<ModuleName>::path-to-template.phtml". Follow this approach will avoid fails render template and easier for developer can find and navigate to file template.
```
Comment (rogyar) [inline src/guides/v2.3/frontend-dev-guide/layouts/xml-manage.md:243]:
```suggestion
The `template` attribute should be declared with the name of the module it belongs to: Example `template="<VendorName>_<ModuleName>::path-to-template.phtml"`. Following this approach will avoid fails upon template rendering and make it easier for the developer to find and navigate to file template.
```

**Comment by rogyar** [comment]
Hi @mrtuvn. Thank you for your contribution. 
Please, check my recommendations above. 

Thank you!

File: src/guides/v2.3/frontend-dev-guide/layouts/xml-instructions.md (hunk 1 above)
Comment (BarnyShergold) [reply src/guides/v2.3/frontend-dev-guide/layouts/xml-instructions.md:82]: This sentence does not scan well in English and is confusing, can you rephrase to make it clearer what it is trying to explain?

File: src/guides/v2.3/frontend-dev-guide/layouts/xml-manage.md (hunk 2 above)
Comment (BarnyShergold) [inline src/guides/v2.3/frontend-dev-guide/layouts/xml-manage.md:243]: Further clarity recommendation
```suggestion
The `template` attribute should be declared with the name of the module it belongs to, eg. `template="<VendorName>_<ModuleName>::path-to-template.phtml"`. Following this approach will avoid failures upon template rendering and make it easier for the developer to find and navigate to the template file.```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]

File: src/guides/v2.3/frontend-dev-guide/layouts/xml-instructions.md (hunk 1 above)
Comment (mrtuvn) [reply src/guides/v2.3/frontend-dev-guide/layouts/xml-instructions.md:82]: In this scope of layout IMO we should not need mention with php way approach ?

File: src/guides/v2.3/frontend-dev-guide/layouts/xml-instructions.md [hunk 3]
```diff
@@ -78,6 +78,9 @@ Sample of usage in the product listing page layout:
 <block class="Magento\Catalog\Block\Product\ListProduct" name="category.products.list" as="product_list" template="Magento_Catalog::product/list.phtml"/>
 ```
 
+{:.bs-callout-warning}
+In layout xml when define new block or re-used exists block with `<block></block>` if `template` attribute not available, so that block will not render by default.
```
Comment (BarnyShergold) [inline src/guides/v2.3/frontend-dev-guide/layouts/xml-instructions.md:82]: Sorry but the English syntax in this sentence is very jumbled. I've read it several times and I'm not sure what it is trying to say. If you could explain it more fully I can then probably help fix the sentence (eg in English you would never follow the word 'when' with a present tense verb and there are no instances of the word 'the' in this sentence and I think there should be at least 4)

Sorry to be strict but documentation needs to be clear and unambiguous.

**Review by BarnyShergold — CHANGES_REQUESTED** [review]

File: src/guides/v2.3/frontend-dev-guide/layouts/xml-instructions.md [hunk 4]
```diff
@@ -65,7 +65,7 @@ We recommend always adding a `name` to blocks. Otherwise, it is given a random n
...
 | `before` | Used to position the block before an element under the same parent. The element name or alias name is specified in the value. Use dash (-) to position the block before all other elements of its level of nesting. See [before and after attributes](#fedg_xml-instrux_before-after) for details. | Element name or dash (-) | no |
 | `after` | Used to position the block after an element under the same parent. The element name or alias name is specified in the value. Use dash (-) to position the block after all other elements of its level of nesting. See [before and after attributes](#fedg_xml-instrux_before-after) for details. | Element name or dash (-) | no |
-| `template` | A template that represents the functionality of the block to which this attribute is assigned. | `Vendor_Module::path/to/template.phtml` (Scope is already in the `templates` directory of the module) | no |
+| `template` | A template that represents the functionality of the block to which this attribute is assigned. If the attribute is omitted, the block will not rendering any output. | `Vendor_Module::path/to/template.phtml` (Scope is already in the `templates` directory of the module) | no |
```
Comment (BarnyShergold) [inline src/guides/v2.3/frontend-dev-guide/layouts/xml-instructions.md:68]:
```suggestion
| `template` | A template that represents the functionality of the block to which this attribute is assigned. If the attribute is omitted, the block will not render any output. | `Vendor_Module::path/to/template.phtml` (Scope is already in the `templates` directory of the module) | no |
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
One small change and it will be perfect.

**Comment by mrtuvn** [comment]
Thanks for all suggestions. I have updated as request in latest commit

**Review by BarnyShergold — APPROVED** [review]

**Review by dobooth — APPROVED** [review]

**Comment by dobooth** [comment]
running tests
